src/PlayFabBuddy.Infrastructure/Adapter/PlayFab/Admin/PlayStreamAdapter.cs

```diff
@@ -0,0 +1,64 @@
+﻿using PlayFab;
+using PlayFab.AdminModels;
+using PlayFabBuddy.Lib.Adapter.Accounts;
+using PlayFabBuddy.Lib.Interfaces.Adapter;
+
+namespace PlayFabBuddy.Infrastructure.Adapter.PlayFab.Admin;
+
+public class PlayStreamAdapter: IPlayStreamAdapter
+{
+    private readonly PlayFabAdminInstanceAPI playFabAdminInstanceApi;
+
+    public PlayStreamAdapter(PlayFabAdminInstanceAPI playFabAdminInstanceApi)
+    {
+        this.playFabAdminInstanceApi = playFabAdminInstanceApi;
+    }
+
+    /// <summary>
+    /// Get all players in a given segment, identified by <paramref name="segmentId"/>
+    /// </summary>
+    /// <param name="segmentId">The segment's ID</param>
+    /// <returns>All the Master Player Accounts in the segment</returns>
+    public async Task<List<MasterPlayerAccountAdapter>> GetPlayersInSegment(string segmentId)
+    {
+        var getPlayersInSegmentRequest = new GetPlayersInSegmentRequest
+        {
+            SegmentId = segmentId
+        };
+        var playersInSegment = await this.playFabAdminInstanceApi.GetPlayersInSegmentAsync(getPlayersInSegmentRequest);
+
+        var accounts = new List<MasterPlayerAccountAdapter>(playersInSegment.Result.ProfilesInSegment);
+        foreach (var profile in playersInSegment.Result.PlayerProfiles)
+        {
+            var account = new MasterPlayerAccountAdapter(profile.PlayerId);
+            accounts.Add(account);
+        }
+
+        return accounts;
+    }
+
+    /// <summary>
+    /// Fetches the ID of the "All Players" segment for the Title
+    /// </summary>
+    /// <returns>The SegmentId</returns>
+    /// <exception cref="Exception">When the segment could not be found</exception>
+    public async Task<string> GetSegmentById(string segmentName)
+    {
+        var allSegmentsResponse = await this.playFabAdminInstanceApi.GetAllSegmentsAsync(new GetAllSegmentsRequest());
+        var allPlayersSegmentId = "";
```

Comment:
`allPlayersSegmentId` are we looking for the "All Players Segment" here or should we have a better name for this?

Reply:
This is a left-over, will fix.